reference/shaders-hlsl/comp/ssbo-array.comp

```diff
@@ -0,0 +1,25 @@
+struct SSBO
+{
+    float4 data[1];
+};
+
+RWStructuredBuffer<SSBO> ssbos : register(u0);
+
+static uint3 gl_GlobalInvocationID;
+struct SPIRV_Cross_Input
+{
+    uint3 gl_GlobalInvocationID : SV_DispatchThreadID;
+};
+
+void comp_main()
+{
+    uint ident = gl_GlobalInvocationID.x;
+    ssbos[1].data[ident] = ssbos[0].data[ident];
```

Comment:
I think it would work for this case:
```
layout(std430, binding = 0) buffer SSBO
{
    vec4 unsized_arr[];
} data;
```

But it is problematic if you have an array of unsized arrays like 
```
layout(std430, binding = 0) buffer SSBO
{
    vec4 unsized_arr[];
} data[10];
```
Because both of these will translate to
```
struct SSBO {
 vec4 unsized_arr[1];
};

RWStructuredBuffer<SSBO> data;
```

The first case is fine because you would just do `data[0].unsized_arr[index]` to access any element of `unsized_arr` and just go "out of bounds". But in the second case, the offsets to `data[1].unsized_arr[0]` and `data[2].unsized_arr[0]` are unknown.